pennylane/__init__.py

```diff
@@ -393,7 +398,15 @@ def run_cnot():
         # Pop the custom decomposition keyword argument; we will use it here
         # only and not pass it to the device.
         custom_decomps = kwargs.pop("custom_decomps", None)
-        decomp_depth = kwargs.pop("decomp_depth", 10)
+        decomp_depth = kwargs.pop("decomp_depth", None)
+
+        if decomp_depth is not None:
+            warnings.warn(
+                "The decomp_depth argument is deprecated and will be removed in version 0.39. ",
+                PennyLaneDeprecationWarning,
+            )
+        else:
+            decomp_depth = 10
```

Comment:
Still wondering if it makes sense to keep this value. While this is consistent with the old behaviour, it will still potentially prevent the decomposition to reach the target gateset of a device. @trbromley would love your input here. Should we change the default `decomp_depth` to keep decomposing until a target gateset is reached or a recursion error occurs, or should we keep default `decomp_depth = 10`?

Reply:
I agree with you that this is a problem, but I as far as I understand deprecations are not about changing behaviour, but just flagging that something will be changed in the future. For now the default behaviour was and still will be (after this PR) for the user to be able to perform this wrong behaviour. Until it is removed in 0.39. That's also why I didn't delete tests that check specifically this behaviour, just marked them to catch the warnings. Still, as you said, would be great to get input from Tom :)

Reply:
@mudit2812 I agree with @EmilianoG-byte that it's ok for old or not-recommended paths to still be available as part of a deprecation cycle before they are removed. Is there a subtlety here that I'm missing?

Reply:
No, I was just wondering.